client/src/app/components/forms/inputs/forms/base-command-form.component.ts

```diff
@@ -19,8 +19,10 @@ require("./form.components.scss");
                         <label>Base Command</label>
                         <label class="secondary-label">What command do you want to call from the image</label>
                         
-                        <base-command-input [inputControl]="baseCommandForm.controls['baseCommand']"
-                                        [baseCommand]="baseCommand"></base-command-input>
+                        <expression-input [inputControl]="baseCommandForm.controls['baseCommand']"
+                                            [expression]="baseCommand">
```

Comment:
Yes, but the expression value here is handled separately from the form control, and projected onto it afterwards in ExpressionInput, so there are 2 value sources, 2-way bound to each other.

You can do 
```typescript
this.baseCommandForm = this.formBuilder.group({
    baseCommand: [this.baseCommand, Validators.compose([Validators.required, Validators.minLength(1)])]
});
```

and have all the necessary information, validation and change detection with https://github.com/rabix/cottontail/pull/74/files/b36954501217df854a6d56de0289c43ca51a7893#diff-516202a22d05b348b2e2ca43d848a963R24